pgmpy/base/_mixin_roles.py

```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+import warnings
```

Comment:
We should use the pgmpy logger from `pgmpy.global_vars import logger`. You can find examples of this in the codebase.